src/jingle_sip/mod_jingle_sip_backend.erl

```diff
@@ -0,0 +1,214 @@
+-module(mod_jingle_sip_backend).
+
+-include("mongoose.hrl").
+
+-type call_id() :: binary().
+-type incoming_request() :: {node(), binary()}.
+-type outgoing_handle() :: binary().
+
+-export([init/2]).
+-export([set_incoming_request/5]).
+-export([set_incoming_handle/2]).
+-export([set_outgoing_request/4]).
+-export([set_outgoing_handle/4]).
+-export([set_outgoing_accepted/1]).
+-export([set_incoming_accepted/1]).
+-export([get_incoming_request/2]).
+-export([get_outgoing_handle/2]).
+-export([get_session_info/2]).
+-export([remove_session/1]).
+
+-record(jingle_sip_session, {sid,
+                             dialog,
+                             state,
+                             direction,
+                             request,
+                             node,
+                             owner,
+                             from,
+                             to,
+                             now,
+                             meta}).
+
+init(_Host, _Opts) ->
+    mnesia:create_table(jingle_sip_session,
+                        [{ram_copies, [node()]},
+                         {attributes, record_info(fields, jingle_sip_session)}]).
+
+-spec set_incoming_request(CallID :: call_id(), ReqID :: binary(),
+                           From :: jid:jid(), To :: jid:jid(), exml:element()) ->
+    ok | {error, any()}.
+set_incoming_request(CallID, ReqID, From, To, JingleEl) ->
+    TFun = pa:bind(fun set_incoming_request_tr/5, CallID, ReqID, From, To, JingleEl),
```

Comment:
pa:bind way of passing arguments would disable some dializer checks.

Compare:

```erlang
    -module(pa_check).  
    -export([call_my_fun/0]).  
    -export([call_pa_fun/0]).                                                              

    call_my_fun() ->                                                                       
        my_fun(1).                                                                         

    call_pa_fun() ->                                                                       
        pa_fun(1).                                                                         

    my_fun(A) ->                                                                         
        mnesia:transaction(fun() -> my_fun_tr(A) end).                                   

    pa_fun(A) ->                                                                         
        mnesia:transaction(pa:bind(fun my_fun_tr/1, A)).                                 

    my_fun_tr(x) ->                                                                      
        ok. 
```

and code

```erlang
    _build/default/lib/mongooseim/src/pa_check.erl                                                                                                                                
      12: The call pa_check:my_fun_tr(A::1) will never return since it differs in the 1st argument from the success typing arguments: ('x')  
```

I don't see why we need it here.